### ISSUE TITLE
Examples of write tiffs from scratch, not inheriting info from an existing tiff

### ISSUE THREAD

**Issue by dbuscombe-usgs** [DigitalSlideArchive/tifftools#55]
Please could you provide more examples for how to use this tool to write tiffs from scratch? Thanks SO much for providing this tool. I really hope I can make it work for  me!

Problem:
- I would like to write out numpy arrays to bigtiff with custom metadata AND exif data
- I'm getting exif data from a jpeg, then generating outputs from that jpeg within a script, that I want to write those outputs to a tiff with exif and metadata

Alternatives tried:
- I cant use tifffile because it doesnt deal with exif data
- I cant use PIL because it doesnt write exif data
- Rasterio doesnt seem to have an exif option
- I cant follow the provided example because I dont have a tiff to read as a starting point

So I'm really hoping I can use tifftools, but I cant figure out (from the one provided example) how to write a tiff entirely from scratch using arrays

Question:
The provided example is useful, but only deals with the case where all the `info` is obtained from an existing file

`info = tifftools.read_tiff(file)`

But how would you create that `info` object from an numpy array, metadata, and exif separately? where metadata and exif are two separate dicts or JSON objects (or whatever)?

I also found [this](https://stackoverflow.com/questions/66887690/changing-exif-data-of-a-tif-file-affects-the-rest-metadata) that helpfully explains how to add info to an *existing* tiff

In summary, I'm looking for advice for how to construct a viable `info` object that can be passed to `.write_tif` like this

`tifftools.write_tif(info, ...)`

where `info` is constructed from 1) a numpy array containing the image, 2) an array or dictionary of exif data, and 3) an array or dictionary of other metadata (tags)

**Comment by manthey** [comment]
Writing image data is outside the scope of `tifftools`.

I would use a library like pyvips to do it (see the `pyvips.Image.new_from_memory` method for converting numpy to vips).  If you need to adjust metadata more than vips allowed, I'd take the output tiff from vips and manipulate it using tifftools or some other tool.

I'm closing this issue as writing image data is beyond what tifftools is ever intended to do.

**Comment by dbuscombe-usgs** [comment]
Ok, fair enough - thanks for responding. Good tip on the `pyvips` package
